Training/Training/file.c

```diff
@@ -1,13 +1,169 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
+// Copyright (c) Gokul TS @Metamation India.
 // ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
+// file.c
+// C Program to implement Bubble sort and binary search algorithm with test cases
 // ------------------------------------------------------------------------------------------------
-
 #include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdlib.h>
+
+#define MAX_DIGIT 13
+#define TEST_SIZE 7
+#define ANSI_RESET_ALL "\x1b[0m"
+#define ANSI_COLOR_RED "\x1b[31m"
+#define ANSI_COLOR_GREEN "\x1b[32m"
+#define ANSI_COLOR_YELLOW "\x1b[33m"
+
+/// <summary>Changes the input array into a sorted array</summary>
+void BubbleSort (int arr[], int size);
+/// <summary>Returns the index of key element in an array</summary>
+int BinarySearch (int sortArr[], int key, int size);
+/// <summary>Gets integer input from user</summary>
+int GetUserInput (char* prompt);
+/// <summary>Implements the search and sort algorithm</summary>
+void SearchAndSort ();
+/// <summary>Clears the content on screen</summary>
+void ClrScreen ();
+/// <summary>Checks test cases</summary>
+void TestCases ();
+/// <summary>Compares two arrays and returns the result</summary>
+int CompareArrays (int* input, int* output, int size);
+
+int main () {
+   while (1) {
+      int num = GetUserInput ("1-SORT AND SEARCH\n2-TEST CASES\n3-CLEAR SCREEN\n4-EXIT\nEnter an option:");
+      if (num > 4 || num < 1)
+         printf (ANSI_COLOR_RED"Please enter a number between 1-4\n"ANSI_RESET_ALL);
+      else {
+         switch (num) {
+            case 1:
+               SearchAndSort ();
+               break;
+            case 2:
+               TestCases ();
+               break;
+            case 3:
+               ClrScreen ();
+               break;
+            case 4:
+               return 0;
+         }
+      }
+   }
+   return 0;
+}
+
+void BubbleSort (int arr[], int size) {
+   for (int i = 0; i < size - 1; i++) {
+      for (int j = 0; j < size - i - 1; j++) {
+         if (arr[j] > arr[j + 1]) {
+            arr[j] = arr[j] + arr[j + 1];
+            arr[j + 1] = arr[j] - arr[j + 1];
+            arr[j] = arr[j] - arr[j + 1];
+         }
+      }
+   }
+}
+
+int BinarySearch (int sortArr[], int key, int size) {
+   int low = 0, high = size - 1, result = -1;
+   while (low <= high) {
+      int mid = low + (high - low) / 2;
+      if (sortArr[mid] == key) {
+         result = mid;
+         high = mid - 1;
+      }
+      if (sortArr[mid] < key) low = mid + 1;
+      else high = mid - 1;
+   }
+   return result;
+}
+
+int GetUserInput (char* prompt) {
+   int64_t number = 0;
```

Comment:
You can keep this number as int itself as after getting the input you are converting it to int. For simplification purpose you can declare the variable as int and work with int values itself.